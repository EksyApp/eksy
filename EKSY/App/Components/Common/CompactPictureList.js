import React, {Component} from 'react'
import {ListView, View, StyleSheet} from 'react-native'
import Image from "./Image";

const imagesPerRow = 4

export default class CompactPictureList extends Component {
	
	constructor(props) {
		super(props)
		
		
		
		const datasource = new ListView.DataSource({rowHasChanged: (r1,r2) => r1 !== r2})
		
		this.state = {
			dataSource: datasource.cloneWithRows(this._makeGroups(this.props.data, imagesPerRow))
		}
		
	}
	
	componentWillReceiveProps(props) {
		if(props.data !== this.props.data) {
			const dataSource = this.state.dataSource.cloneWithRows(this._makeGroups(props.data, imagesPerRow) || []);
			this.setState({dataSource})
		}
	}
	
	_makeGroups(data, itemsPerGroup) {
		let groups = []
		for (var i = 0; i < data.length; i += itemsPerGroup) {
			let group = []
			for (var j = i; j < Math.min(data.length, i+itemsPerGroup); j++) {
				group.push(data[j])
			}
			groups.push(group)
		}
		return groups
	}
	
	renderImageRow(data) {
		return(
				<View style={styles.rowContainer}>
					{data.map((datum, index) => {
						return(
								<View style={styles.imageContainer} key={index}>
									<Image source={{uri: datum.uri}} style={styles.image} />
								</View>
						)
					})}
				</View>
		)
	}
	
	render() {
		return(
				<ListView
						style = {this.props.listStyle}
						dataSource={this.state.dataSource}
						renderRow = {(data) => this.renderImageRow(data)}
				/>
		)
	}
	
}

const styles = StyleSheet.create({
	rowContainer: {
		height: 100,
		flexDirection: 'row'
	},
	
	imageContainer: {
		width: 100/imagesPerRow + "%",
		justifyContent: 'center',
		height: '100%',
	},
	
	image: {
		width: '100%',
		height: '100%',
		resizeMode: 'cover'
	}
})