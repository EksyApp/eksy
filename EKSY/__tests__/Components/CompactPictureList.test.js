import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CompactPictureList from "../../App/Components/Common/CompactPictureList";

test('CompactPictureLit renders correctly', () => {
	const tree = renderer.create(
			<CompactPictureList
					data={[]}
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});