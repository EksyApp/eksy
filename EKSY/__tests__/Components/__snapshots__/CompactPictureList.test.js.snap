// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactPictureLit renders correctly 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 0,
    }
  }
  enableEmptySections={true}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={undefined}
>
  <View />
</RCTScrollView>
`;
